src/edi_energy_scraper/__init__.py

```diff
@@ -301,6 +303,27 @@ async def _download(
             raise
         return file_path
 
+    @staticmethod
+    def get_edifact_format(path: Path) -> Tuple[EdifactFormatVersion, List[Optional[EdifactFormat]]]:
+        """
+        Determines the edifact formats and the version of a given file.
+        A file can describe more than one format (for example APERAK and CONTRL).
+        Therefore, a list of all formats described in a file is returned.
+        """
+        filename = path.stem
+        date_string = filename.split("_")[-1]  # Assuming date is in the last part of filename
+        date_format = "%Y%m%d"
+        berlin = pytz.timezone("Europe/Berlin")
+        berlin_local_time = datetime.datetime.strptime(date_string, date_format).astimezone(berlin)
+        version = get_edifact_format_version(berlin_local_time)
+        edifactformat: List[Optional[EdifactFormat]] = []
+        for entry in EdifactFormat:
+            if str(entry) in filename:
+                edifactformat.append(entry)
+        if not edifactformat:
+            edifactformat = [None]
+        return version, edifactformat
```

Comment:
```suggestion
        edifactformats: List[Optional[EdifactFormat]] = []
        for entry in EdifactFormat:
            if str(entry) in filename:
                edifactformats.append(entry)
        if not any(edifactformats):
            edifactformats = [None]
        return version, edifactformats
```
plural dinge (wie listen) sollen plural namen haben

Reply:
ahja, oder so. Wie gesagt, nur plural S überlese ich leicht